lib/promscrape/client.go

```diff
@@ -154,7 +153,7 @@ func (c *client) ReadData(dst []byte) ([]byte, error) {
 	if statusCode != fasthttp.StatusOK {
 		metrics.GetOrCreateCounter(fmt.Sprintf(`vm_promscrape_scrapes_total{status_code="%d"}`, statusCode)).Inc()
 		return dst, fmt.Errorf("unexpected status code returned when scraping %q: %d; expecting %d; response body: %q",
-			c.scrapeURL, statusCode, fasthttp.StatusOK, dst[dstLen:])
+			c.scrapeURL, statusCode, fasthttp.StatusOK, dst)
```

Comment:
Could you pls bring more context to this change? Why it is needed?

Reply:
Should we update dstLen on L:143 instead?

Reply:
yep, it can be fixed by recalculating dstLen at L:143, but in the future it may lead to another bug with slice out of bound access. 
 As far as i see, dstLen  limits output for decompressed result.

Reply:
what do you think about sanity check before error fmt ?
```go
if len(dst) < dstLen {
   dstLen = len(dst)
}
```

Reply:
Oh, that was just error reporting... Sorry, my bad!